src/method.rs

```diff
@@ -0,0 +1,309 @@
+    ///
+    /// ```edition2021
+    /// use serde::{Deserialize, Serialize};
+    /// use wmi::{COMLibrary, Variant, WMIConnection, WMIResult};
+    ///
+    /// #[derive(Serialize)]
+    /// #[serde(rename_all = "PascalCase")]
+    /// struct CreateInput {
+    ///     command_line: String
+    /// }
+    ///
+    /// #[derive(Deserialize)]
+    /// #[serde(rename_all = "PascalCase")]
+    /// struct CreateOutput {
+    ///     return_value: u32,
+    ///     process_id: u32
+    /// }
+    ///
+    /// #[derive(Deserialize)]
+    /// #[serde(rename = "Win32_Process")]
+    /// struct Win32Process;
+    ///
+    /// fn main() -> WMIResult<()> {
+    ///
+    ///     let wmi_con = WMIConnection::new(COMLibrary::new()?)?;
+    ///
+    ///     // Note: The Create call can be unreliable, so consider using another means of starting processes.
+    ///     let input = CreateInput {
+    ///         command_line: "systeminfo".to_string()
+    ///     };
+    ///     let output: CreateOutput = wmi_con.exec_class_method::<Win32Process, _, _>("Create", input)?;
+    ///
+    ///     println!("The return code of the Create call is {}", output.return_value);
+    ///     println!("The ID of the created process is: {}", output.process_id);
+    ///
+    ///     Ok(())
+    /// }
+    /// ```
+    pub fn exec_class_method<MethodClass, In, Out>(
+        &self,
+        method: impl AsRef<str>,
+        in_params: In,
+    ) -> WMIResult<Out>
+    where
+        MethodClass: de::DeserializeOwned,
+        In: Serialize,
+        Out: de::DeserializeOwned,
+    {
+        let (method_class, _) = struct_name_and_fields::<MethodClass>()?;
+        self.exec_instance_method::<MethodClass, In, Out>(method, method_class, in_params)
+    }
+
+    /// Executes a WMI method on a specific instance of a class. Examples include
+    /// [GetSupportedSize](https://learn.microsoft.com/en-us/windows-hardware/drivers/storage/msft-Volume-getsupportedsizes) of `MSFT_Volume`
+    /// and [Pause](https://learn.microsoft.com/en-us/windows/win32/cimwin32prov/pause-method-in-class-win32-printer) of `Win32_Printer`.
+    ///
+    /// `MethodClass` should have the name of the class on which the method is being invoked.
+    /// `In` and `Out` can be `()` or any custom structs supporting (de)serialization containing the input and output parameters of the function.
+    /// `object_path` is the `__PATH` variable of the class instance on which the method is being called, which can be obtained from a WMI query.
+    ///
+    /// Arrays, Options, unknowns, and nested objects cannot be passed as input parameters due to limitations in how variants are constructed by `windows-rs`.
+    ///
+    /// ```edition2021
+    /// use serde::{Deserialize, Serialize};
+    /// use wmi::{COMLibrary, FilterValue, Variant, WMIConnection, WMIResult};
+    ///
+    /// #[derive(Deserialize)]
+    /// #[serde(rename_all = "PascalCase")]
+    /// struct PrinterOutput {
+    ///     return_value: u32
```

Comment:
Same w.r.t naming: I think sticking with the Windows convention is more consistent.
Can you also note that `ReturnValue` is a generic value that's expected to always be returned by `ExecMethod`?